src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -194,13 +197,23 @@ private void FileWorkItemsHelper(SarifLog sarifLog, SarifWorkItemContext filingC
 
                 this.FiledWorkItems.AddRange(task.Result);
 
-                // TODO: We need to process updated work item models to persist filing
-                //       details back to the input SARIF file, if that was specified.
-                //       This code should either return or persist the updated models
-                //       via a property, so that the file work items command can do
-                //       this work.
-                //
-                //       https://github.com/microsoft/sarif-sdk/issues/1774
+                foreach (Run run in sarifLog.Runs)
+                {
+                    if (run.Results == null) { continue; }
+
+                    foreach (Result result in run.Results)
+                    {
+                        result.WorkItemUris ??= new List<Uri>();
+                        result.WorkItemUris.Add(workItemModel.HtmlUri);
```

Comment:
>. [](start = 30, length = 1)

It's non-obvious that this is mutating the original sarif file (at least I presume it is.) You might add a comment.

Reply:
+1 done.

---
In reply to: [398143633](https://github.com/microsoft/sarif-sdk/pull/1829#discussion_r398143633) [](ancestors = 398143633)